Init/#1 settings

dain391 opened the pull request:
## Issue Number  
> closed #1 

## 📝 Description  
- MSA 기반 프로젝트 초기 세팅 완료  
- 전체 구조: `common-lib`, `user-service`, `hub-service`, `order-service`, `gateway-service`, `eureka-server`  
- 공통 모듈(`common-lib`) 구성  
  - `ApiResponse`, `PageResponse` (공통 응답/페이징 구조)  
  - `ErrorCode`, `CustomException`, `GlobalExceptionHandler` (글로벌 예외 처리)  
  - `BaseTimeEntity` (엔티티 공통 시간 필드)  
- `.env.example` 및 `.gitignore` 정리 (`application-local.yml` 제외 처리)  
- 각 서비스별 `application.yml` 환경 및 JPA/Auditing/Eureka 설정 추가  
- `build.gradle`, `settings.gradle` 기반 멀티모듈 연결 완료  

## 🌐 Test Result  
- 모든 서비스(`Eureka`, `Gateway`, `User`, `Hub`, `Order`) 정상 기동 확인  
<img width="1536" height="771" alt="image" src="https://github.com/user-attachments/assets/38e4e292-41d1-4c38-b306-a61b890f3565" />

## 🔎 To Reviewer  
- 공통 예외 처리 및 응답 구조 일관성 확인 부탁드립니다. 
- 각 서비스 DB 설정 및 Eureka 연결 정상 여부 확인 바랍니다.
- 헬스체크 추가하면 좋을지 의견 부탁드립니다!

dain391 committed:
chore: .env.example 추가 및 .gitignore 정리

- .env.example 템플릿 생성
- .gitignore 업데이트
- 코드 스타일 추가

#1

dain391 committed:
build: MSA 멀티모듈 골격 추가

- settings.gradle에 모듈 include(eureka/gateway/user/hub/order/common-lib)
- 루트 build.gradle에 Spring Boot/Cloud BOM 및 공통 설정(subprojects) 적용
- 각 모듈 build.gradle 기본 의존성 구성
- 서비스별 application.yml 초안 및 포트/프로필 세팅

#1

dain391 committed:
feat: BaseTimeEntity 추가

- BaseTimeEntity(createdAt/updatedAt/deletedAt, markAsDeleted) 추가

#1

dain391 committed:
feat: 글로벌 예외 처리 및 공통 응답 구조 추가

- ApiResponse 클래스 추가 (성공/실패 응답 통일)
- CustomException, ErrorCode, GlobalExceptionHandler 구현
- 공통 예외 및 검증 에러 처리 로직 정리
- 각 서비스에서 예외 발생 시 ApiResponse 형태로 일관된 응답 반환

#1

dain391 committed:
feat: 공통 페이징 응답 구조 추가

- PageResponse<T> 클래스 추가
- Spring Data Page 및 커스텀 리스트 결과를 통일된 형태로 변환 가능
- content, totalElements, totalPages, page, size, sort 필드 포함

#1

GoodNyong commented:
확인했습니다!

dyun23 reviewed (dismissed):
너무 다 잘해주셔서 바로 프로젝트 진행해도 좋을 것 같습니다☺️☺️

sonaanweb commented on common-lib/src/main/java/com/oneforlogis/common/model/BaseTimeEntity.java:
by, isdeleted boolean 같은 필드가 빠져있는데 BaseEntity가 아닌 TimeEntity로만 관리하는 이유가 혹시 따로 있을까요?

dain391 replied:
말씀하신 isDeleted boolean은 deletedAt != null로도 soft delete 여부를 판별할 수 있어서
중복 필드로는 두지 않았습니다!

isDeleted boolean 추가하고 BaseEntity로 변경하는게 나을까요?

Sp-PJS replied:
소나님 의견대로 createdBy, updatedBy, deletedBy(누구에 의해 생성, 수정, 삭제되었는지에 대한 필드ex) 해당 username or user_id가 값으로 들어가는 필드) 필드가 필요할 거 같습니다! 발제자료 요구사항에도 나와있어요

sonaanweb replied:
> 말씀하신 isDeleted boolean은 deletedAt != null로도 soft delete 여부를 판별할 수 있어서 중복 필드로는 두지 않았습니다!
> 
> isDeleted boolean 추가하고 BaseEntity로 변경하는게 나을까요?

네! 우선 userid by 필드랑 꼭 boolean으로 관리되지 않아도 deletedat 판별 자체도 이 파일 내에서 관리하는 게 좋을 거 같습니다!

` ex) public boolean isDeleted() {
        return deletedAt != null;
    }`

Sp-PJS reviewed (dismissed):
확인했습니다 다만 ApiResponse나 ErrorCode에 몇가지 더 추가되거나 변경사항이 있을 거 같습니다!

dain391 committed:
feat: Auditing 기능 및 AuditorAware 설정 추가

- BaseEntity에 createdAt, updatedAt, deletedAt 등 기본 시간 필드 추가
- createdBy, updatedBy, deletedBy 감사자 필드 및 deleted 플래그 추가
- markAsDeleted(), restore() 유틸 메서드 구현
- user/hub/order 서비스별 JpaAuditConfig 설정
  - user-service: SecurityContext 기반 감사자 추적
  - hub/order-service: X-Username 헤더 기반 감사자 추적

dain391 committed:
feat: ApiResponse record 클래스 수정 및 ResponseEntity import 오류 해결

- ApiResponse 헬퍼 메서드(success, created, accepted, noContent) 구조 개선
- HttpStatus 기반 생성자 추가로 코드 일관성 확보
- Swagger @Schema 어노테이션으로 API 응답 필드 문서화
